### PR TITLE
Added Profile pages MVP

### PR DESCRIPTION
**Pull request by vocatwk**
Each profile page is now able to display users' sent messages and About Me sections(as well giving users the ability to edit it). I have attached screenshots below. The second one is when the user is logged in and looking at their own profile. The first is when the user isn't either of those things. I had to edit User.java to allow storing and editing AboutMe messages and persistentDataStore.java in order to store and retrieve aboutMe messages from DataStore. I've also added a style for the messages in main.css.

screen shot 2018-05-22 at 8 22 59 pm
screen shot 2018-05-22 at 8 20 39 pm

### CODE REVIEW

File: src/main/java/codeu/model/data/User.java [hunk 1]
```diff
@@ -23,6 +23,7 @@
   private final String name;
   private final String passwordHash;
   private final Instant creation;
+  private String aboutMe = null;
```
Comment (lealgamb) [inline src/main/java/codeu/model/data/User.java]: Maybe set to null in the constructor.

[the remark this answers is not in the document]
Comment (vocatwk) [reply src/main/java/codeu/model/data/User.java]: oh I see, okay. I will make that change.

File: src/main/java/codeu/controller/ProfileServlet.java [hunk 2]
```diff
@@ -87,8 +90,32 @@ public void doGet(HttpServletRequest request, HttpServletResponse response)
...
+
+    if(username == null){
+      // user not logged in
+      response.sendRedirect("/login");
+      return;
+    }
+
+    if(!username.equals(subject.getName())){
+      // user is trying to edit another user's profile
+      // TODO respond with access denied
```
[the remark this answers is not in the document]
Comment (vocatwk) [reply src/main/java/codeu/controller/ProfileServlet.java]: will do.

File: src/main/java/codeu/controller/ProfileServlet.java [hunk 3]
```diff
@@ -87,8 +90,32 @@ public void doGet(HttpServletRequest request, HttpServletResponse response)
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response)
       throws IOException, ServletException {
-  	
-	// TO-DO
+    String requestUrl = request.getRequestURI();
+
+	  String username = (String) request.getSession().getAttribute("user");
```
[the remark this answers is not in the document]
Comment (vocatwk) [reply src/main/java/codeu/controller/ProfileServlet.java]: will correct.

File: src/main/java/codeu/model/data/User.java (hunk 1 above)
Comment (vocatwk) [reply src/main/java/codeu/model/data/User.java]: You're right. I will change that.

**Comment by vocatwk** [comment]
I had to add a null check in PersistenDataStore.java because users added before the aboutMe change won't have an aboutMe parameter and so if it's not set(null) I set it to an empty string in memory.

**Pull request merged**